### ISSUE TITLE
Should empty class ignores type checking

### ISSUE THREAD

**Issue by vivekgalatage** [Microsoft/TypeScript#26708]
**TypeScript Version:**  3.1.0

**Search Terms:** Empty Class, Type Checking

**Code**
1. Empty Child class
```ts
// Notice the Child class is an empty one
class Child {
}

class ChildrenManager {
    public addChild(alarm: Child): boolean {
        return true;
    }
}

let manager = new ChildrenManager();

// The below line doesn't throw any error
manager.addChild(manager);
```

[Playground 1](http://www.typescriptlang.org/play/#src=class%20Child%20%7B%0D%0A%7D%0D%0A%0D%0Aclass%20ChildrenManager%20%7B%0D%0A%20%20%20%20public%20addChild(alarm%3A%20Child)%3A%20boolean%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20true%3B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Alet%20manager%20%3D%20new%20ChildrenManager()%3B%0D%0Amanager.addChild(manager)%3B)

2. Non-empty Child class
```ts
class Child {
    doSomething(): void {
    }
}

class ChildrenManager {
    public addChild(alarm: Child): boolean {
        return true;
    }
}

let manager = new ChildrenManager();

// Now the error is reported for non-empty child class for the type
manager.addChild(manager);
```
[Playground 2](http://www.typescriptlang.org/play/#src=class%20Child%20%7B%0D%0A%20%20%20%20doSomething()%3A%20void%20%7B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Aclass%20ChildrenManager%20%7B%0D%0A%20%20%20%20public%20addChild(alarm%3A%20Child)%3A%20boolean%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20true%3B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Alet%20manager%20%3D%20new%20ChildrenManager()%3B%0D%0Amanager.addChild(manager)%3B)

**Expected behavior:**
Type checks should not depend on whether the class is empty or not.
In both cases error should be thrown like
```
Argument of type 'ChildrenManager' is not assignable to parameter of type 'Child'.
```
**Actual behavior:**
Class being empty or not, the type check depends on it.

**Comment by RyanCavanaugh** [comment]
The template instructs you to read the FAQ; please do!

![image](https://user-images.githubusercontent.com/6685088/44734528-fe221000-aa9e-11e8-8665-ca43077b9430.png)
